src/new/wallet.js

```diff
@@ -1122,15 +1201,21 @@ class HathorWallet extends EventEmitter {
    */
   async onEnterStateProcessing() {
     // Started processing state now, so we prepare the local data to support using this facade interchangable with wallet service facade in both wallets
-    return this.storage.processHistory()
-      .then(() => { this.setState(HathorWallet.READY); })
-      .catch(() => { this.setState(HathorWallet.ERROR); });
+    try {
+      await this.processTxQueue();
+      this.setState(HathorWallet.READY);
+    } catch (e) {
+      this.setState(HathorWallet.ERROR);
+    }
   }
 
   setState(state) {
     if (state === HathorWallet.PROCESSING && state !== this.state) {
       // XXX: will not await this so we can process history on background.
-      this.onEnterStateProcessing();
+      this.onEnterStateProcessing().catch(e => {
+        console.error(e);
+        this.setState(HathorWallet.ERROR);
```

Comment:
The `onEnterStateProcessing()` already handles the wallet state, right?
Should we repeat this `setState()` command here?

Reply:
This is just a safeguard agains unhandled errors, since we use unawaited promises I think we should at least enforce this.